### PR TITLE
fix(node-canvas): imageSmoothing setter

### PR DESCRIPTION
**Pull request by ShaMan123**
fixes https://github.com/fabricjs/fabric.js/issues/8030
override default `patternQuality` from `good` to `best` for node canvas
https://github.com/Automattic/node-canvas/blob/0bccc05b384cc1f3e969cd9ed7859a95376ec88c/src/CanvasRenderingContext2d.cc#L1479

Should we expose a way to control `patternQuality`? Should it be set on if imageSmoothing is enabled?

Seems this prop doesn't exist on the browser. The only thing I found that comes near is https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/imageSmoothingQuality

### CODE REVIEW

File: src/static_canvas.class.js [hunk 1]
```diff
@@ -755,6 +755,7 @@
       this.calcViewportBoundaries();
       this.clearContext(ctx);
       fabric.util.setImageSmoothing(ctx, this.imageSmoothingEnabled);
+      ctx.patternQuality = 'best';
```
Comment (ShaMan123) [inline src/static_canvas.class.js:758]: Maybe fast as default?

Comment (ShaMan123) [reply src/static_canvas.class.js:758]: fast isn't good enough
look at the diff between `best` and `fast`
image

Comment (ShaMan123) [reply src/static_canvas.class.js:758]: objectsInActiveSelections


objectsInActiveSelections

**Comment by asturur** [comment]
is a node-canvas non standard property, that control some cairo parameters.
I m fine leaving it in a way that satisfy most expectations, but i don't want to add logic around that ( when to set best or fast )

**Review by asturur — APPROVED** [review]

**Pull request merged**